WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsSettingsHelper.kt

```diff
@@ -0,0 +1,68 @@
+package org.wordpress.android.ui.bloggingprompts
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.map
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.runBlocking
+import org.wordpress.android.fluxc.store.BloggingRemindersStore
+import org.wordpress.android.ui.mysite.SelectedSiteRepository
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.util.DateUtils.isSameDay
+import org.wordpress.android.util.config.BloggingPromptsEnhancementsFeatureConfig
+import org.wordpress.android.util.config.BloggingPromptsFeatureConfig
+import java.util.Date
+import javax.inject.Inject
+
+class BloggingPromptsSettingsHelper @Inject constructor(
+    private val bloggingRemindersStore: BloggingRemindersStore,
+    private val selectedSiteRepository: SelectedSiteRepository,
+    private val appPrefsWrapper: AppPrefsWrapper,
+    private val bloggingPromptsFeatureConfig: BloggingPromptsFeatureConfig,
+    private val bloggingPromptsEnhancementsFeatureConfig: BloggingPromptsEnhancementsFeatureConfig,
+) {
+    fun getPromptsCardEnabledLiveData(
+        siteId: Int
+    ): LiveData<Boolean> = bloggingRemindersStore.bloggingRemindersModel(siteId)
+        .asLiveData()
+        .map { it.isPromptsCardEnabled }
+
+    fun updatePromptsCardEnabledBlocking(siteId: Int, isEnabled: Boolean) = runBlocking {
+        updatePromptsCardEnabled(siteId, isEnabled)
+    }
+
+    @Suppress("MemberVisibilityCanBePrivate")
+    suspend fun updatePromptsCardEnabled(siteId: Int, isEnabled: Boolean) {
+        val current = bloggingRemindersStore.bloggingRemindersModel(siteId).firstOrNull() ?: return
+        bloggingRemindersStore.updateBloggingReminders(current.copy(isPromptsCardEnabled = isEnabled))
+    }
+
+    fun isPromptsFeatureAvailable(): Boolean {
+        val selectedSite = selectedSiteRepository.getSelectedSite() ?: return false
+        return bloggingPromptsFeatureConfig.isEnabled() && selectedSite.isPotentialBloggingSite
+    }
+
+    suspend fun isPromptsFeatureActive(): Boolean {
```

Comment:
To me `Available` (`isPromptsFeatureAvailable`) and `Active` (`isPromptsFeatureActive`) sound the same in this context. I was trying to think of a name suggestion, but couldn't think of one 🤔

Reply:
Yeah, I got those confused more than once while developing. I could only think of the following:
- `isPromptsFeatureAvailable` remains the same
- `isPromptsFeatureActive` -> `shouldShowPromptsFeature`

Wdyt?

Reply:
Makes sense to me 👍